channels/channel.nos/nos2010/chn_nos2010.py

```diff
@@ -1331,6 +1332,81 @@ def update_video_item_live(self, item):
         return item
 
+    def create_iptv_streams(self, parameter_parser):
+        """ Fetch the available live channels using EPG endpoint and format them into JSON-STREAMS
+
+        :param ActionParser parameter_parser: a ActionParser object to is used to parse and
+                                                   create urls
+
+        :return: Formatted stations
+        :rtype: list
+        """
+
+        epg_data = JsonHelper.loads(UriHandler.open(datetime.datetime.now().strftime("https://start-api.npo.nl/epg/%Y-%m-%d?type=tv"),
+            no_cache=True,
+            additional_headers=self.__jsonApiKeyHeader))
+
+        parent_item = MediaItem("Live", "https://www.npostart.nl/live", media_type=mediatype.FOLDER)
+        items = []
+        iptv_streams = []
+
+        for stations in epg_data["epg"]:
+            livestream = stations["channel"]["liveStream"]
+            item = MediaItem(livestream["title"], livestream["shareUrl"], media_type=mediatype.VIDEO)
+            items.append(item)
+
+            iptv_streams.append(dict(
+                id=livestream["id"],
+                name=livestream["title"],
+                logo=livestream["images"]["original"]["formats"]["tv"]["source"],
+                group=self.channelName,
+                stream=parameter_parser.create_action_url(self, action=action.PLAY_VIDEO, item=item, store_id=parent_item.guid),
+            ))
+        
+        parameter_parser.pickler.store_media_items(parent_item.guid, parent_item, items)
+
+        return iptv_streams
+
+    def create_iptv_epg(self, parameter_parser):
+        """ Fetch the EPG using the EPG endpoint and format it into JSON-EPG
+
+        :param ActionParser parameter_parser: a ActionParser object to is used to parse and
+                                                   create urls
+
+        :return: Formatted stations
+        :rtype: dict
+        """
+
+        parent = MediaItem("EPG", "https://start-api.npo.nl/epg/", media_type=mediatype.FOLDER)
+        iptv_epg = dict()
+        media_items = []
+
+        start = datetime.datetime.now() - datetime.timedelta(days=3)
+        for i in range(0, 7, 1):
+            air_date = start + datetime.timedelta(i)
+            data = UriHandler.open(air_date.strftime("https://start-api.npo.nl/epg/%Y-%m-%d?type=tv"),
+                no_cache=True,
+                additional_headers=self.__jsonApiKeyHeader)
+
+            json_data = JsonHelper.loads(data)
+            for epg_item in json_data["epg"]:
+                id = epg_item["channel"]['liveStream']['id']
+                iptv_epg[id]=iptv_epg.get(id, [])
+                for program in epg_item["schedule"]:
+                    media_item = MediaItem(program["program"]["title"], program["program"]["id"], media_type=mediatype.EPISODE)
+                    media_items.append(media_item)
+                    iptv_epg[id].append(dict(
+                        start=JsonHelper.safeget(program, "startsAt"),
```

Comment:
No `safeget` but `get_value` here.

Reply:
That one makes it hard to keep it tidy. 
We cannot construct a new JsonHelper object, once you have parsed it. This makes it hard to iterate over the lists the Pythonic way.

What solution would you prefer?

1.
I have seen this format
 ```python
dummy = JsonHelper("{}")
dummy.json = episodes_json
```
2. Allow get_value to return a JsonHelper-object as well
3. Check in constructor if JsonHelper `data` might already be a `list` or `dict`, so that we can also wrap already parsed JSON's
4. Iterate over the keys only and stick to the parent JsonHelper. I have tried that, but that makes it really ugly.

Reply:
I understand. Then I would suggest to create a static method `get_from`:

```python
@staticmethod
def get_from(data, *args, logger=None, **kwargs):
    """ Retrieves data from a generci JSON object based on the input parameters
    
    :param str args|int:    The dictionary keys, or list indexes.
    :param any kwargs:      Possible value = fallback and allows the specification of a fallback value.
    
    :return: the selected JSON object
    
    """
```
The use the code from `get_value` and get rid of the `data = self.json` part and replace the `self.logger` for just `logger`. Finally let the `get_value` call the  `get_from` with `self.json` as `data` and inject the `self.logger` als `logger=`.

Does that make sense?